mono/metadata/marshal-ilgen.c

```diff
@@ -5949,8 +5949,9 @@ emit_marshal_object_ilgen (EmitMarshalContext *m, int argnum, MonoType *t,
 			break;
 		}
 
-		if (t->attrs & PARAM_ATTRIBUTE_OUT) {
-			mono_mb_emit_byte (mb, CEE_LDNULL);
+		if ((t->attrs & (PARAM_ATTRIBUTE_IN | PARAM_ATTRIBUTE_OUT)) == PARAM_ATTRIBUTE_OUT) {
+			mono_mb_emit_byte (mb, MONO_CUSTOM_PREFIX);
```

Comment:
Why is this needed ?

Reply:
Well a parameter can be both in and out (I've added one to the tests). If it's an In parameter, it requires input conversion, regardless of whether it's also an Out parameter or not. The related test simply flips the sign of the inputs, so this is expected.

For output-only parameters, no input conversion is required (they can be zero-initialized), but we still need to create an object, where the output will go to, even if it's not initialized. Otherwise it will generate a NullReference exception. Of course without this change the tests will generate such an exception.